### ISSUE TITLE
Autotune Quantizing Keyboard Input.

### ISSUE THREAD

**Issue by ca3games** [LMMS/lmms#5976]
### Enhancement Summary
It would be a feature that uses the ghost note channel and when you record your MIDI keyboard playing, It automatically tries to snap the notes to the nearest note in the ghost channel, also It would Quantize your notes you're playing to a % (which you can configure) off your keyboard being played.
Similar to how "inking" works on drawing apps, like Krita.

#### Justification
It would allow non keyboard players, like me, that have 0 experience playing piano, to just "Quantize" a melody in a superior way, even if your playing piano skills are subpar.
Basically you be like "inking" the piano notes with a drawing tablet if you were drawing on Krita.
Would be basically a superior way to "hummanize" in a easy way MIDI notes using a MIDI keyboard.

#### Mockup
Just a button next to the record button or ghost notes, where some red border appears on the MIDI piano ghost roll notes and the notes get filled after you play them with the piano roll, and anything outside to these "red ghost border" notes would be not drawabled.

**Comment by qnebra** [comment]
Isn't source of ghost notes be midi file, with that melody perfectly done? Unless author of midi screw something. This feature sounds to me like kind of piano play training app.

**Comment by ca3games** [comment]
> Isn't source of ghost notes be midi file, with that melody perfectly done? Unless author of midi screw something. This feature sounds to me like kind of piano play training app.

No, I meant the ghost notes are used to quantize the notes that are being played while you record a MIDI instrument.

**Comment by qnebra** [comment]
But if you had in mind making multiple takes, then there is something like copy and paste. Or using ghost notes as visual indicator of your melody. Or practise a lot and record with metronome. Or record with rest of music piece playing alongside.

**Comment by ca3games** [comment]
> But if you had in mind making multiple takes, then there is something like copy and paste. Or using ghost notes as visual indicator of your melody. Or practise a lot and record with metronome. Or record with rest of music piece playing alongside.

Yes, but this is meant to be a feature for people without strong keyboard skills.

**Comment by qnebra** [comment]
Then record it with whatever skill user had, and tweak melody using avalaible tools. It could be a lot of editing, but completly doable with any keyboard skill. I see your point here, with that feature.

**Comment by ca3games** [comment]
> Then record it with whatever skill user had, and tweak melody using avalaible tools. It could be a lot of editing, but completly doable with any keyboard skill. I see your point here, with that feature.

Yes, but this will simplify a lot of the effort for people without playing skills.
Think of this as the "autotune" of the piano roll for MIDI input.

**Comment by qnebra** [comment]
But, something needs to be source of ghost notes:

1. previously recorded version of that melody, this mean user already can play that melody perfectly fine, maybe just needed some practice
2. from existing midi file, then it could be a thing (I use midi to learn how to play when needed, usually begins on sequence, later figuring out tempo and length). But still, why not just copy & paste melody?

**Comment by ca3games** [comment]
> But, something needs to be source of ghost notes:
> 
> 1. previously recorded version of that melody, this mean user already can play that melody perfectly fine, maybe just needed some practice
> 2. from existing midi file, then it could be a thing (I use midi to learn how to play when needed, usually begins on sequence, later figuring out tempo and length). But still, why not just copy & paste melody?

I just mentioned the idea is for a more "manual" way to "humanize" some MIDI recording.
My idea is that you can get some MIDI instrument, like a keyboard and have automated while you play all the velocity and CC MIDI signals you would create with that instrument, which would be dificult or cumbersome to create with just mouse + keyboard.

This is why I initially mentioned a custom % of distance to allow for the human player to be offsync and create that more natural and realistic feeling to the notes.

**Comment by ca3games** [comment]
Realized there's already a record quantization now, closing this issue.
